### DISCUSSION TITLE
email placeholders for event start, end and admission dates

### DISCUSSION THREAD

**Discussion by listerr** [pretix/pretix#3007] [Ideas]
- Add email placeholders for event start, and and admission dates, e.g.` {event_date_from} {event_date_to} {event_admission_time}`

I like to send out reminders for events with other details a week or so before. Currently there appears to be no way
to include the event start/end/admission time, so we have to write email templates with it manually added like this:

```
Dear {name},

We are looking forward to seeing you at our upcoming event:

{event}

October 17 - 18th 2022

Admission will start at 9:30 on both days.

... etc.

```

Looking at the source code, it looks slightly more complicated because events can have a series or subevents, e.g.

https://github.com/pretix/pretix/blob/a568a37f4c2344e1a967ddf02d954c577de13a70/src/pretix/base/templates/pretixbase/email/order_details.html

Has some logic:

```
        <tr>
            <td>
                <strong>{% trans "Event:" %}</strong>
            </td>
            <td>
                {{ event.name }}
                {% if not event.has_subevents and event.settings.show_dates_on_frontpage %}
                    <br>
                    {{ event.get_date_range_display }}
                    {% if event.settings.show_times %}
                        {{ event.date_from|date:"TIME_FORMAT" }}
                    {% endif %}
                {% endif %}
            </td>
        </tr>
```

I'm not sufficiently skilled in python to know how to add new placeholders to https://github.com/pretix/pretix/blob/ce177227c74f3ee32c513a3b230bacd3e6c7767f/src/pretix/base/email.py

But looks like dates are used in other places there.

**Comment by raphaelm** [comment]
Yes, subevents ("event series" in the interface) have been the reason we never added this so far. It's less that the implementation is hard, it's that the UX would be horrible.

What should the placeholder do if an order contains tickets for multiple different dates?

**Reply by listerr** [reply]
Looking at the source gave me a hint that this is not so straightforward.

In the email design templates, it _does_ try to do this, so the event name, date etc. appear at the bottom of all outgoing emails, so it's a bit odd that this can't be included in the email text itself.

I suppose for subevents, it could look like:-

- Use the dates from the next event the customer is booked to attend. 
  (e.g. send email to everyone, but `{event_date_from}` would be set to the next date they have booked.)

- If a date range is selected, Use the dates from the next event the customer is booked to attend within the given date range.

Or: placeholder just contains a list (e.g. comma separated) of all the dates/date(s) found within the given date range. Might look a little ugly if there were lots of dates.
